packages/builder/src/pages/builder/app/[application]/data/datasource/[datasourceId]/index.svelte

```diff
@@ -1,163 +1,89 @@
-  import { datasources, integrations, queries, tables } from "stores/backend"
-  import EditDatasourceConfig from "./_components/EditDatasourceConfig.svelte"
-  import RestExtraConfigForm from "components/backend/DatasourceNavigator/TableIntegrationMenu/rest/RestExtraConfigForm.svelte"
-  import PlusConfigForm from "components/backend/DatasourceNavigator/TableIntegrationMenu/PlusConfigForm.svelte"
+  import { Tabs, Tab, Heading, Body, Layout } from "@budibase/bbui"
+  import { datasources, integrations } from "stores/backend"
   import ICONS from "components/backend/DatasourceNavigator/icons"
-  import CapitaliseRenderer from "components/common/renderers/CapitaliseRenderer.svelte"
-  import { IntegrationTypes } from "constants/backend"
-  import { isEqual } from "lodash"
-  import { cloneDeep } from "lodash/fp"
-  import ImportRestQueriesModal from "components/backend/DatasourceNavigator/modals/ImportRestQueriesModal.svelte"
-  import Spinner from "components/common/Spinner.svelte"
-
-  const querySchema = {
-    name: {},
-    queryVerb: { displayName: "Method" },
-  }
-
-  let importQueriesModal
-  let changed = false
-  let isValid = true
-  let integration, baseDatasource, datasource
-  let queryList
-  let loading = false
+  import EditDatasourceConfig from "./_components/EditDatasourceConfig.svelte"
+  import TablesPanel from "./_components/panels/Tables/index.svelte"
+  import RelationshipsPanel from "./_components/panels/Relationships.svelte"
+  import QueriesPanel from "./_components/panels/Queries/index.svelte"
+  import RestHeadersPanel from "./_components/panels/Headers.svelte"
+  import RestAuthenticationPanel from "./_components/panels/Authentication/index.svelte"
+  import RestVariablesPanel from "./_components/panels/Variables/index.svelte"
 
-  $: baseDatasource = $datasources.selected
-  $: queryList = $queries.list.filter(
-    query => query.datasourceId === datasource?._id
-  )
-  $: hasChanged(baseDatasource, datasource)
-  $: updateDatasource(baseDatasource)
+  let selectedPanel = null
+  let panelOptions = []
+  $: integration = datasource && $integrations[datasource.source]
 
-  const hasChanged = (base, ds) => {
-    if (base && ds) {
-      changed = !isEqual(base, ds)
-    }
-  }
+  $: datasource = $datasources.selected
 
-  const saveDatasource = async () => {
-    try {
-      // Create datasource
-      await datasources.update({ datasource, integration })
-      if (datasource?.plus) {
-        await tables.fetch()
-      }
-      await datasources.fetch()
-      notifications.success(`Datasource ${name} updated successfully.`)
-      baseDatasource = cloneDeep(datasource)
-    } catch (err) {
-      notifications.error(`Error saving datasource: ${err}`)
-    }
-  }
+  $: getOptions(datasource)
 
-  const updateDatasource = base => {
-    if (base) {
-      datasource = cloneDeep(base)
-      integration = $integrations[datasource.source]
+  const getOptions = datasource => {
+    if (datasource.plus) {
```

Comment:
When deleting a datasource, this throws: 

`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'plus')` 

Just needs a null check for the deleted state.

Reply:
Ah good catch, didn't test the deletion flow